src/main/kotlin/no/nav/tilleggsstonader/sak/vilkår/stønadsperiode/StønadsperiodeService.kt

```diff
@@ -27,20 +31,24 @@ class StønadsperiodeService(
     }
 
     @Transactional
-    fun lagreStønadsperioder(behandlingId: BehandlingId, stønadsperioder: List<StønadsperiodeDto>): List<StønadsperiodeDto> {
-        validerBehandling(behandlingId)
+    fun lagreStønadsperioder(
+        behandlingId: BehandlingId,
+        stønadsperioder: List<StønadsperiodeDto>,
+    ): List<StønadsperiodeDto> {
+        val behandling = behandlingService.hentSaksbehandling(behandlingId)
+        validerBehandling(behandling)
         validerStønadsperioder(behandlingId, stønadsperioder)
 
         val tidligereStønadsperioder = stønadsperiodeRepository.findAllByBehandlingId(behandlingId)
 
-        slettStønadsperioder(tidligereStønadsperioder, stønadsperioder)
-        val oppdaterteStønadsperioder = oppdaterEksisterendeStønadsperioder(tidligereStønadsperioder, stønadsperioder)
-        val nyeStønadsperioder = leggTilNyeStønadsperioder(behandlingId, stønadsperioder, tidligereStønadsperioder)
+        slettStønadsperioder(behandling, tidligereStønadsperioder, stønadsperioder)
```

Comment:
Ikke relevant for denne PR-en men vi burde kjøre en rename på denne metoden til noe mer forklarende. Ala. `slettStønadsperioderSomErFjerna` eller noe liknende